File: 189-rotate-array/rotate-array.java
class Solution {
    private void reverser(int[] arr, int start, int end) {
        while (start < end) {
            int temp = arr[end];
            arr[end] = arr[start];
            arr[start] = temp;
            start++;
            end--;
        }
    }

    public void rotate(int[] nums, int k) {
        k=k%nums.length;
        reverser(nums,0,nums.length-1);
        reverser(nums,0,k-1);
        reverser(nums,k,nums.length-1);
    }
}
// -1,-100,3,99
// -100,3,99,-1

// 1,2,3,4,5,6,7

File: 189-rotate-array/rotate-array.py
class Solution:
    def reverse(self, nums, start, end):
        while start < end:
            nums[start], nums[end] = nums[end], nums[start]
            start += 1
            end -= 1

    def rotate(self, nums: List[int], k: int) -> None:
        """
        Do not return anything, modify nums in-place instead.
        """
        if len(nums)<2:
            return
        k=k%len(nums)
        self.reverse(nums, 0, len(nums) - 1)
        self.reverse(nums, k, len(nums) - 1)
        self.reverse(nums, 0, k - 1)
